community/kernel/src/main/java/org/neo4j/kernel/impl/api/store/LegacyStorePropertyCursor.java

```diff
@@ -104,31 +102,124 @@ public Object value()
     @Override
     public boolean booleanValue()
     {
-        return (Boolean) value();
-    }
+        Object value = value();
+        if ( value instanceof Boolean )
+        {
+            return (Boolean) value;
+        }
+        else
+        {
+            throw new IllegalStateException( "Value is not a boolean" );
+        }    }
```

Comment:
Also use generic method that takes a `Class<T>` and probably pull it to some helper class